### PR TITLE
LibJS: Updates to Intl.DurationFormat

### PR DESCRIPTION
**Pull request by trflynn89**
```
Diff Tests:
    test/intl402/DurationFormat/prototype/format/invalid-arguments-throws.js        ❌ -> ✅
    test/intl402/DurationFormat/prototype/formatToParts/invalid-arguments-throws.js ❌ -> ✅
```

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Intl/DurationFormat.cpp [hunk 1]
```diff
@@ -141,43 +141,66 @@ ThrowCompletionOr<Temporal::DurationRecord> to_duration_record(VM& vm, Value inp
...
+        auto value = TRY(input_object.get(name));
+        if (value.is_undefined())
+            return false;
 
-    // 4. For each row of Table 1, except the header row, in table order, do
-    for (auto const& duration_instances_component : duration_instances_components) {
-        // a. Let valueSlot be the Value Slot value of the current row.
-        auto value_slot = duration_instances_component.value_slot;
+        value_slot = TRY(Temporal::to_integer_if_integral(vm, value, ErrorType::TemporalInvalidDurationPropertyValueNonIntegral, name, value));
+        return true;
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/Intl/DurationFormat.cpp:159]: Alternatively you could set `any_defined = true` here in this case but this works as well

File: Userland/Libraries/LibJS/Tests/builtins/Intl/DurationFormat/DurationFormat.prototype.formatToParts.js [hunk 2]
```diff
@@ -266,7 +266,11 @@ describe("correct behavior", () => {
 
 describe("errors", () => {
     test("non-object duration records", () => {
-        [-100, Infinity, NaN, "hello", 152n, Symbol("foo")].forEach(value => {
+        expect(() => {
+            new Intl.DurationFormat().formatToParts("hello");
+        }).toThrowWithMessage(RangeError, "is not an object");
+
+        [-100, Infinity, NaN, 152n, Symbol("foo")].forEach(value => {
```
Comment (davidot) [inline Userland/Libraries/LibJS/Tests/builtins/Intl/DurationFormat/DurationFormat.prototype.formatToParts.js:273]: For completeness (although this is not directly relevant to this PR) you could add booleans, null and undefined but definitely not needed

**Review by davidot — APPROVED** [review]
Two tiny not needed suggestions but LGTM

File: Userland/Libraries/LibJS/Runtime/Intl/DurationFormat.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibJS/Runtime/Intl/DurationFormat.cpp]: Yeah that's much nicer

**Review by davidot — APPROVED** [review]

**Pull request merged**